common/src/main/java/com/wynntils/features/user/WorldWaypointDistanceFeature.java

```diff
@@ -0,0 +1,136 @@
+import net.minecraft.client.Camera;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class WorldWaypointDistanceFeature extends UserFeature {
+
+    @Config
+    public CustomColor textColor = CommonColors.WHITE;
+
+    @Config
+    public int backgroundOpacity = 80;
+
+    @Config
+    public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.NONE;
+
+    private Vector3d ndc = null;
+    private String distanceText;
+
+    @Override
+    public List<Class<? extends Model>> getModelDependencies() {
+        return List.of(CompassModel.class);
+    }
+
+    @SubscribeEvent
+    public void onRenderLevelPost(RenderLevelEvent.Post event) {
+        if (CompassModel.getCompassLocation().isEmpty()) return;
+
+        Location location = CompassModel.getCompassLocation().get();
+        Matrix4f projection = event.getProjectionMatrix().copy();
+        Camera camera = event.getCamera();
+        Vec3 cameraPos = camera.getPosition();
+
+        // apply camera rotation
+        projection.multiply(Vector3f.XP.rotationDegrees(camera.getXRot()));
+        projection.multiply(Vector3f.YP.rotationDegrees(camera.getYRot() + 180.0F));
+
+        // offset to put text to the center of the block
+        float dx = (float) (location.x + 0.5 - cameraPos.x);
+        float dy = (float) (location.y + 0.5 - cameraPos.y);
+        float dz = (float) (location.z + 0.5 - cameraPos.z);
+
+        double squaredDistance = dx * dx + dy * dy + dz * dz;
+
+        double distance = Math.sqrt(squaredDistance);
+        int maxDistance = McUtils.mc().options.renderDistance * 16;
+
+        this.distanceText = Math.round((float) distance) + "m";
+
+        // move the position to avoid ndc z leak past 1
+        if (distance > maxDistance) {
+            double posScale = maxDistance / distance;
+            dx *= posScale;
+            dz *= posScale;
+        }
+
+        worldToNdc(new Vector3f(dx, dy, dz), projection);
+    }
+
+    @SubscribeEvent
+    public void onRenderGuiPost(RenderEvent.Post event) {
+        if (CompassModel.getCompassLocation().isEmpty() || ndc == null || isOnScreen()) return;
+
+        Window window = event.getWindow();
+
+        float backgroundWidth = FontRenderer.getInstance().getFont().width(distanceText);
+        float backgroundHeight = FontRenderer.getInstance().getFont().lineHeight;
+        WynntilsMod.info(String.valueOf(backgroundWidth));
+
+        float displayPositionX = (float) ((ndc.x + 1.0f) / 2.0f) * window.getGuiScaledWidth();
+        float displayPositionY = (float) ((1.0f - ndc.y) / 2.0f) * window.getGuiScaledHeight();
+
+        RenderUtils.drawRect(
+                event.getPoseStack(),
+                CommonColors.BLACK.withAlpha(backgroundOpacity),
+                displayPositionX - (backgroundWidth / 2) - 2,
+                displayPositionY - (backgroundHeight / 2) - 2,
+                0,
+                backgroundWidth + 3,
```

Comment:
Shouldn't you do + 2 to be in-line with every other alignment?

Reply:
![image](https://user-images.githubusercontent.com/60712627/201423766-2562fc0f-d030-4520-974d-c2aefcb83359.png)
+2 doesn't seems to be enough space, make the m looks like it's suffocating, not sure what's the cause of this, maybe kerning?

Reply:
We probably lose 1 pixel in some floating point to int conversion or smth. I guess keep it.